src/v/kafka/server/group_metadata.cc

```diff
@@ -201,26 +201,32 @@ void offset_metadata_value::encode(
   response_writer& writer, const offset_metadata_value& v) {
     writer.write(v.version);
     writer.write(v.offset);
-    writer.write(v.leader_epoch);
+    if (v.version >= group_metadata_version{3}) {
+        writer.write(v.leader_epoch);
+    }
     writer.write(v.metadata);
     writer.write(v.commit_timestamp);
+    if (v.version == group_metadata_version{1}) {
+        writer.write(v.expiry_timestamp);
```

Comment:
Question, could there ever be an issue with compatibility across versions because we were always deserializing this int64 value but never serializing it?

Reply:
not really since we only deserializing this for version == 1